cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```diff
@@ -79,15 +82,17 @@ def _get_file_tag_fetcher(delivery_type: DataDelivery) -> FetchDeliveryFileTagsS
             DataDelivery.FASTQ: SampleAndCaseDeliveryTagsFetcher,
             DataDelivery.ANALYSIS_FILES: SampleAndCaseDeliveryTagsFetcher,
             DataDelivery.FASTQ_ANALYSIS: SampleAndCaseDeliveryTagsFetcher,
+            DataDelivery.BAM: BamDeliveryTagsFetcher,
```

Comment:
Adding a whole new delivery flow by just adding a few lines in this file is so nice!